PeopleCounterApi/Message/Commands/PeopleCounter/PeopleCounterUpdateCommand.cs

```
@@ -1,45 +1,49 @@
+using PeopleCounterApi.ViewModels;
+using PeopleCounterCore.Interfaces.Commands;
+using PeopleCounterCore.Interfaces.Services;
+using PeopleCounterCore.UnitOfWork;
+using PeopleCounterInfrastructure;
+using PeopleCounterInfrastructure.Repositories;
+using System.Threading.Tasks;
+using PeopleCounterCore.Services;
 
-//namespace PeopleCounterApi.Message.Commands.PeopleCounter
-//{
-//    public class PeopleCounterUpdateCommand : ICommand, ICommandFactory
-//    {
-//        public string Name => "peopleCounter.update";
-//        private RabbitMessageModel _messageModel;
-//        private IPeopleCounterService _peopleCounterService;
+namespace PeopleCounterApi.Message.Commands.PeopleCounter
+{
+    public class PeopleCounterUpdateCommand : ICommand, ICommandFactory
+    {
+        public string Name => "peopleCounter.update";
+        private RabbitMessageModel _messageModel;
+        private IPeopleCounterService _peopleCounterService;
 
-//        public async Task Execute()
-//        {
-//            PeopleCounterViewModel peopleCounterView = JsonConvert.DeserializeObject<ViewModels.PeopleCounterViewModel>(_messageModel.Message);
-//            PeopleCounterCore.Models.PeopleCounter peopleCounter = Mapping.MapPeopleCounter(peopleCounterView);
+        public async Task Execute()
+        {
+            PeopleCounterGetViewModel peopleCounterView = JsonConvert.DeserializeObject<ViewModels.PeopleCounterGetViewModel>(_messageModel.Message);
+            PeopleCounterCore.Models.PeopleCounter peopleCounter = Mapping.MapPeopleCounter(peopleCounterView);
+            peopleCounter.Id = 0;
+            await _peopleCounterService.UpdatePeopleCounter(peopleCounter);
+            if (peopleCounter.Id == 0)
+            {
+                throw new ApplicationException($"PeopleCounter with ID: {peopleCounterView.Id} for shop {peopleCounterView.ShopId} could not be added, PeopleCounter received from Rabbit was: {JsonConvert.SerializeObject(_messageModel.Message)}");
```

Don't let it propagate...